### PR TITLE
Cato 4003

### PR DESCRIPTION
**Pull request by mario-paniccia**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/abridged/accountsApproval/AccountsApproval.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        validateApproverRequired(boxRetriever),
+        validateAtMost12Approvers(boxRetriever),
+        validateAtMost12OtherApprovers(boxRetriever),
+        validateApprovers(boxRetriever),
+        validateOtherApprovers(boxRetriever)
+      )
+    }
+  }
+
+  def validateApproverRequired(boxRetriever: AbridgedAccountsBoxRetriever)(): Set[CtValidation] = {
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/abridged/accountsApproval/AccountsApproval.scala]: Could these functions be private?

File: src/main/scala/uk/gov/hmrc/ct/box/CtValidation.scala [hunk 2]
```diff
@@ -18,7 +18,9 @@ package uk.gov.hmrc.ct.box
 
 import play.api.libs.json.Json
 
-case class CtValidation(boxId: Option[String], errorMessageKey: String, args:Option[Seq[String]]=None)
+case class CtValidation(boxId: Option[String], errorMessageKey: String, args:Option[Seq[String]] = None) {
+  def isGlobalError = !boxId.isDefined
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/box/CtValidation.scala]:
```
def isGlobalError = boxId.isEmpty
```

File: src/main/scala/uk/gov/hmrc/ct/box/Validators.scala [hunk 3]
```diff
@@ -78,6 +78,17 @@ trait Validators {
 
   protected def isEmpty(value: Option[_]): Boolean = value.isEmpty
 
+  protected def collectWithBoxId(newBoxId: String) (errors: Set[CtValidation]): Set[CtValidation] = {
+    val (globalErrors, boxErrors) = errors.partition(_.isGlobalError)
+    val transformedBoxErrors = boxErrors.map (error => error.copy(boxId = Some(newBoxId)))
+    globalErrors ++ transformedBoxErrors
+  }
+
+  protected def replaceBoxId(newBoxId: String, errors: () => Set[CtValidation])(): Set[CtValidation] = {
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/box/Validators.scala]: Remove the braces?

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/abridged/accountsApproval/AccountsApproval.scala (hunk 1 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/abridged/accountsApproval/AccountsApproval.scala]: yep

File: src/main/scala/uk/gov/hmrc/ct/box/CtValidation.scala (hunk 2 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/box/CtValidation.scala]: ok

File: src/main/scala/uk/gov/hmrc/ct/box/Validators.scala (hunk 3 above)
Comment (mario-paniccia) [inline src/main/scala/uk/gov/hmrc/ct/box/Validators.scala]: ok

**Pull request merged**